webapp/packages/core-utils/src/MetadataMap.ts

```diff
@@ -97,7 +97,12 @@ export class MetadataMap<TKey, TValue> implements Map<TKey, TValue> {
       }
 
       const value = provider(key, this);
-      this.temp.set(key, observable(value as any));
+
+      if (typeof value === 'object' && value !== null) {
+        this.temp.set(key, observable(value));
+      }
+
+      this.temp.set(key, value);
```

Comment:
this will lead to double set and final value will be always observer, probably you forgot to `return`